src/Text/Layout/Table.hs

```diff
@@ -232,14 +232,17 @@ tableLines :: Cell a
            -> [RowGroup a] -- ^ Rows which form a cell together
            -> [String]
 tableLines specs TableStyle { .. } header rowGroups =
-    topLine : addHeaderLines (rowGroupLines ++ [bottomLine])
+    maybe id (:) topLine . addHeaderLines $ maybe id (\b -> (++[b])) bottomLine rowGroupLines
   where
     -- Helpers for horizontal lines that will put layout characters arround and
     -- in between a row of the pre-formatted grid.
 
-    -- | Generate columns filled with 'sym'.
-    fakeColumns sym
-                  = map (`replicateCharB` sym) colWidths
+    -- | Generate columns filled with 'sym', or blank spaces if 'sym' is of width 0.
+    fakeColumns sym' = map replicateSym colWidths
+      where
+        replicateSym w = concat $ replicate q sym ++ [take r sym]
+          where (q,r) = w `quotRem` l
+        (sym,l) = let l' = length sym' in if l' == 0 then (" ",1) else (sym',l')
```

Comment:
* Please always use a space after a comma.
* Put definitions within a where on a new line.  (With `let .. in ..` and `if` I do it sometimes as well if it improves readability, which it usually does, but I leave it up to you about those.)